harmonica/forward/point_mass.py

```diff
@@ -249,6 +257,32 @@ def kernel_g_z_cartesian(easting, northing, upward, easting_p, northing_p, upwar
     return (upward - upward_p) / distance ** 3
 
 
+@jit(nopython=True)
+def kernel_g_northing_cartesian(
+    easting, northing, upward, easting_p, northing_p, upward_p
+):
+    """
+    Kernel function for northing component of gravity gradient in Cartesian coordinates
+    """
+    distance = distance_cartesian(
+        [easting, northing, upward], [easting_p, northing_p, upward_p]
+    )
+    return (northing - northing_p) / distance ** 3
```

Comment:
There's a little sign mistake on this line, but I think I triggered it when I wrote the `kernel_g_z_cartesian`.

Because the acceleration vector points towards the point source (if it has positive mass), this line should be:
```suggestion
    return - (northing - northing_p) / distance ** 3
```

Consider a point mass located on the origin and two computation points located at 10 meters on the northing direction, but located at opposite sides of the point source. The `g_northing` on the computation point located on the positive axe of `northing` should be negative, while `g_northing` on the other one should positive (and both with the same absolute value).
So, the following code should give the following result:
```python
import harmonica as hm
coordinates = ([0, 0], [-10, 10], [0, 0])
point_mass = [0, 0, 0]
masses = [100, 100]
hm.point_mass_gravity(
    coordinates, point_mass, masses, field="g_northing", coordinate_system="cartesian"
)
>>> array([6.673e-06,  -6.673e-06])
```

I think this mistake is my fault because I haven't added a comment on `kernel_g_z_cartesian` explaining why I omit the minus sign. The reason for this omission is that `g_z` represent the downward component, although our vertical axes points upwards. Therefore, we must remove the minus sign in order to get the opposite component of the acceleration vector.

I think it would be nice to check the sign of the acceleration components on the symmetry tests (not only for the new `g_easting` and `g_northing`, but also for the `g_z` symmetry) and would be nice to add a little comment on the `kernel_g_z_cartesian` function making very clear that we are intentionally omitting the minus sign. What do you think? Would you like to add these two things on this PR?

Reply:
Ok @santisoler , I will fix these problems. They were my mistake. I will also include a comment in the gz about this signal convention.

Reply:
In the tests for verifying the signals of the components `g_z`, `g_northing` and `g_easting`, may I use simple `assert field < 0`, `assert > 0` and `assert == 0` commands? I did not find routines to do this kind of assertion in the `numpy.testing`.

Reply:
The `assert field < 0` and `assert field > 0` are valid! We try to avoid the `assert field == 0` because on some numerical computations the result is very close to zero but not necessarily strictly equal to zero. So, you can use `assert field == 0` only if we are sure that the computed result will be strictly zero. For example, results from analytical solutions tend to satisfy this requirement, but on numerical approximations having a result strictly equal to zero is very rare.

Nevertheless, on this case I would like to take into account the sign of the mass instead of hardcoding the expected sign of the field. So, for example, if you put a single point mass with positive mass, then the g_z value above it should be positive, so we can assert if the sign of the g_z matches the sign of the mass:

```python
assert np.sign(mass) == np.sign(g_z)
```

If the computation point is now bellow the point mass, then the `g_z` should have the opposite sign of the mass:


```python
assert np.sign(mass) == - np.sign(g_z)
```

Do you agree?

Reply:
Perfect!

Reply:
@birocoles to check if a float is close to 0, use `numpy.testing.assert_allclose(number, 0)`

Reply:
I think tests abotu signal are ok now!

Reply:
I have also included a test for comparing numerical derivatives of Cartesian `potential` and the analytical components given by `g_z`, `g_northing` and `g_easting`. Everything is ok for the horizontal components. There is a problem in `g_z`. During this test, I verified that `kernel_potential_cartesian` was using `distance_spherical` instead of `distance_cartesian`. I fixed this problem, but the problem with the test is still there.